src/service/message.js

```diff
@@ -24,62 +49,89 @@ const findAll = () => {
   return [];
 };
 
-const findOne = id => {
+const findOne = messageId => new Promise((resolve, reject) => {
   if (fs.existsSync('log.txt')) {
-    const fileData = fs.readFileSync('log.txt', 'utf8');
+    const rl = createInterface();
 
-    if (fileData) {
-      const message = fileData.split('\n')[id];
+    rl.on('line', line => {
+      if (line) {
+        const { id, message } = JSON.parse(line);
 
-      if (message) {
-        return message;
+        if (id === parseInt(messageId, 10)) {
+          resolve(message);
+          rl.close();
+          rl.removeAllListeners();
+        }
       }
-    }
-  }
+    });
 
-  return null;
-};
+    rl.on('close', () => {
+      reject(new Error('Message has not been found'));
+    });
+  } else {
+    reject(new Error('file not exiest'));
+  }
+});
 
+const deleteMessage = messageId => new Promise((resolve, reject) => {
+  let lineBreak = '';
 
-const deleteMessage = id => {
   if (fs.existsSync('log.txt')) {
-    const fileData = fs.readFileSync('log.txt', 'utf8');
-
-    if (fileData) {
-      const messages = fileData.split('\n');
-
-      if (id >= messages.length) { throw new Error('Message has not been found'); }
-
-      messages.splice(id, 1);
-
-      fs.writeFileSync('log.txt', messages.join('\n'));
-
-      return;
-    }
+    const rl = createInterface();
+
+    rl.on('line', line => {
+      if (line) {
+        const { id, message } = JSON.parse(line);
+
+        if (id !== parseInt(messageId, 10)) {
+          fs.appendFileSync('log1.txt', `${lineBreak}${JSON.stringify({ id, message })}`);
+        } else {
+          resolve();
+        }
+        if (!lineBreak) { lineBreak = '\n'; }
+      }
+    });
+
+    rl.on('close', () => {
+      fs.unlinkSync('log.txt');
+      fs.renameSync('log1.txt', 'log.txt');
+      reject(new Error('Message has not been found'));
+    });
+  } else {
+    reject(new Error('file not exiest'));
```

Comment:
User shouldn't care if file exists or not. We should send `Message has not been found`.